lib/core/src/sdk.rs

```diff
@@ -344,67 +344,117 @@ impl LiquidSdk {
-            match chain_swap.direction {
-                Direction::Incoming => {
-                    let swap_script = chain_swap.get_lockup_swap_script()?.as_bitcoin_script()?;
-                    let current_height =
-                        self.bitcoin_chain_service.lock().await.tip()?.height as u32;
-                    let locktime_from_height =
-                        LockTime::from_height(current_height).map_err(|e| {
-                            PaymentError::Generic {
-                                err: format!(
-                                    "Error getting locktime from height {current_height:?}: {e}",
-                                ),
-                            }
-                        })?;
-
-                    info!("Checking Chain Swap {} expiration: locktime_from_height = {locktime_from_height:?},  swap_script.locktime = {:?}", chain_swap.id, swap_script.locktime);
-                    if swap_script.locktime.is_implied_by(locktime_from_height) {
-                        let id: &String = &chain_swap.id;
-                        info!("Chain Swap {} user lockup tx was broadcast. Setting the swap to refundable.", id);
-                        self.chain_swap_state_handler
-                            .update_swap_info(id, Refundable, None, None, None, None)
-                            .await?;
-                    }
-                }
-                Direction::Outgoing => {
-                    let swap_script = chain_swap.get_lockup_swap_script()?.as_liquid_script()?;
-                    let current_height = self.liquid_chain_service.lock().await.tip().await?;
-                    let locktime_from_height = elements::LockTime::from_height(current_height)?;
-
-                    info!("Checking Chain Swap {} expiration: locktime_from_height = {locktime_from_height:?},  swap_script.locktime = {:?}", chain_swap.id, swap_script.locktime);
-                    if utils::is_locktime_expired(locktime_from_height, swap_script.locktime) {
-                        self.chain_swap_state_handler
-                            .refund_outgoing_swap(chain_swap)
-                            .await?;
-                    }
-                }
-            }
+    /// Tries refunding a chain swap:
+    /// ## Outgoing Chain Swap
+    /// For outgoing chain swaps, we check whether the status is `RefundPending` and
+    /// the locktime has expired. If so, we refund non-cooperatively; else we refund cooperatively
+    ///
+    /// ## Incoming Chain Swap
+    /// For incoming chain swaps, we check whether the status is `Pending` and the locktime has expired.
+    /// If so, we set the swap status as `Refundable`
+    async fn try_chain_swap_refund(&self, chain_swap: &ChainSwap) -> Result<()> {
+        if chain_swap.user_lockup_tx_id.is_none() || chain_swap.refund_tx_id.is_some() {
+            return Err(anyhow::anyhow!(
+                "Lockup_tx not present or refund_tx already broadcast"
```

Comment:
I think this will produce errors in the logs while the refund tx is unconfirmed. This could be for a while for a Bitcoin tx specially if a low fee rate is selected